core/src/main/java/com/amplifyframework/storage/options/StorageOptions.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.options;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.amplifyframework.core.async.Options;
+import com.amplifyframework.storage.StorageAccessLevel;
+
+/**
+ * Storage options interface requires that every
+ * storage operation will inspect the provided options
+ * instance for access level and target ID.
+ */
+public abstract class StorageOptions implements Options {
```

Comment:
I think this could be non-public, since only the sub classes get used outside of this package. Right?

Reply:
ah shoot you are right. I'm pretty sure I fixed it before, but not sure why its public again now.

Reply:
OH it's because I added the bit for S3RequestUtils... I'll just get rid of that since I don't plan on using it just yet.